SSH: prefer using `runuser` instead of `login`

afonsofrancof opened the pull request:
## Describe your changes

Some distros use util-linux's `login` binary, which doesn't work with NetBird's SSH.
Debian, for example, started using this binary recently in Debian 13, which broke NetBird's SSH.
`runuser` is the recommended utility for this use case and is now tried first on all Linux systems.
If `runuser` doesn't exist, it falls back to the standard login method.
I also removed the special handling for arch-linux, since it just uses `runuser` for that distro now.

## Issue ticket number and link

#4869

## Stack

<!-- branch-stack -->

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/netbirdio/netbird?pullRequest=4873) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/netbirdio/netbird?pullRequest=4873) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/netbirdio/netbird?pullRequest=4873) it.</sub>

afonsofrancof commented:
Changed it :)

afonsofrancof commented:
@alexmoras I changed the PR to always use runuser if available on linux systems.
`runuser` is the recommended utility for this use case, not `login`.
If `runuser` isn't available, `login` is used.

I know this works on recent versions, but could you try it on your Ubuntu 24.04 again, just to make sure there are no regressions? Sorry for all the trouble. (I squashed some commits, so you might have to reclone or delete the commits and then pull from my branch)

afonsofrancof commented:
I can confirm that it doesn't have any regression on Debian 12 (which uses shadow-login)

afonsofrancof commented:
I think the way to do this is by using `unix.Exec`, which replaces the current process with the new command (be it `login`, `su`, etc)
That way, if the command returns, we know it failed (because the only way to continue executing the code after the `unix.Exec` is if it fails).

I think this is how tailscale does it, and if it works for them, I think the same approach should work here.
